File: code_learning/牛客/contest/2024-7-17/题解/F.cpp
#include "bits/stdc++.h"
using namespace std;
// 时间复杂度是：n*log_2(n)，因为求解过程可以看作 n * 调和级数，所以暴力求解即可
int main()
{
    int n;
    cin >> n;
    long long sum = 0;
    vector<int> a(n + 1);
    for (int i = 1; i <= n; i++)
    {
        cin >> a[i];
        sum ^= a[i];
    }
    long long ans = 0;
    for (int i = 1; i <= n; i++)
        for (int j = 1; j <= n / i; j++)
        {
            ans = max(ans, sum ^ a[i] ^ (a[i] + j));
            ans = max(ans, sum ^ a[i] ^ (1ll * a[i] * j));
        }
    cout << ans << '\n';
    return 0;
}

File: code_learning/牛客/contest/2024-7-17/F.py
import sys
from functools import reduce
input = lambda: sys.stdin.readline().rstrip()

ans = 0
n = int(input())
a = [0] + list(map(int, input().split()))
xor_sum = reduce(lambda x, y: x ^ y, a)
# 时间复杂度：sum(n//i) from 1 to n 等价于 n * logn
for i in range(1, n + 1):
    for j in range(1, n // i + 1):
        ans = max(ans,
                  xor_sum ^ a[i] ^ (a[i] + j),
                  xor_sum ^ a[i] ^ (a[i] * j),
                  )

sys.stdout.write(str(ans))